tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -141,30 +208,111 @@ StatusOr<bool> AllReduceReassociate::Run(
       if (!ReassociateAllReduceIsProfitable(ar0, ar1, inst)) {
         continue;
       }
-      if (!AreCompatible(ar0, ar1, *kind)) {
+
+      HloInstruction* convert0 = nullptr;
+      HloInstruction* convert1 = nullptr;
+      if (!MatchOperandsToAllReduceWithOptionalConvert(inst, &convert0,
+                                                       &convert1)) {
+        VLOG(2) << "One or both inputs are type-converted.";
+      }
+      // Check to see if input converts are present and preserving values and
+      // precision.
+      bool should_promote_ar = convert0 || convert1;
+      if (should_promote_ar) {
+        if (!reassociate_converted_ar_) {
+          VLOG(2) << "Inputs have Converts but "
+                     "xla_gpu_enable_reassociation_for_converted_ar is set to "
```

Comment:
nit: can you change this to not mention xla_gpu_enable_reassociation_for_converted_ar flag? For other backends, this flag is not used. Just say promotions of all-reduces for reassociation is disabled.

Reply:
I see, updated here